pkg/app/zipper/tldcache.go

```diff
@@ -34,12 +45,68 @@ func (app *App) probeTopLevelDomains(ms *PrometheusMetrics) {
 	}
 }
 
+type tldPrefix struct {
+	prefix  string
+	nsCount int
+}
+
+func sortedByNsCount(prefixes []string) []string {
+	countedPrefixes := make([]tldPrefix, len(prefixes))
+	for i, prefix := range prefixes {
+		if prefix == "" {
+			countedPrefixes[i] = tldPrefix{
+				prefix:  prefix,
+				nsCount: 0,
+			}
+		}
+		countedPrefixes[i] = tldPrefix{
+			prefix:  prefix,
+			nsCount: strings.Count(prefix, ".") + 1,
+		}
+	}
+	sort.Slice(countedPrefixes, func(i, j int) bool {
+		return countedPrefixes[i].nsCount < countedPrefixes[j].nsCount
+	})
+	sortedPrefixes := make([]string, len(prefixes))
+	for i := range countedPrefixes {
+		sortedPrefixes[i] = countedPrefixes[i].prefix
+	}
+	return sortedPrefixes
+}
+
+func getBackendsForPrefix(prefix string, backends []backend.Backend, tldCache map[string][]*backend.Backend) []*backend.Backend {
```

Comment:
Imho this func is doing something unexpected. It's called getBackendsForPrefix, yet when there is no match it's not returning empty array. Maybe getBackendsForPrefixOrAll might be more intuitive 🤔